src/etc/rc.bootup

```diff
@@ -126,10 +126,11 @@ if (mwexec("/bin/kenv -q pfSense.boot 2>/dev/null") != 0) {
 			continue;
 		}
 
-		@file_put_contents("{$g['cf_conf_path']}/vendor_mac_{$if}",
-		    $if_details['macaddr']);
+		$vendor_mac_arr[$if] = $if_details['macaddr'];
 	}
-	unset($ifs, $if);
+	$vendor_mac_contents = json_encode($vendor_mac_arr);
+	@file_put_contents("{$g['vardb_path']}/vendor_mac", $vendor_mac_contents);
```

Comment:
It's not necessary to set a variable for a single use, you can call json_encode direct on file_put_contents();